### PR TITLE
RandomCommand - Extension

### PR DESCRIPTION
**Pull request by DeadlyFirex**
# Pull request

 - [x] I have read and agreed to the [code of conduct](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [GitHub projects](https://github.com/orgs/CascadeBot/projects).
 - [x] I have tested all of my changes.
 
## Added/Changed
What did you add and/or change for this.

 - `;random color/choice`
 - Fix for PurgeUtils (`smh`)
 - Brand-new ColorUtils
 - Fixed star imports in ColorCommand (smh)

## Feature
If this is a new feature, why should we add it to the bot.
- Because it's fun!

### CODE REVIEW

**Comment by DeadlyFirex** [comment]
You thought you were done @weeryan17 / @binaryoverload ?
Hell no

**Comment by DeadlyFirex** [comment]
Btw, both arguments.json and en_GB.json are outdated due to the recent changes by #176 
I am updating

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.cascadebot.cascadebot.commandmeta.ICommandMain;
+import org.cascadebot.cascadebot.commandmeta.ISubCommand;
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.messaging.MessageType;
+import org.cascadebot.cascadebot.messaging.MessagingObjects;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.utils.ColorUtils;
+import org.cascadebot.cascadebot.utils.RandomUtils;
+
+import java.awt.*;
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java]: 👀

File: src/main/java/org/cascadebot/cascadebot/utils/ColorUtils.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            context.getTypedMessaging().replyDanger(context.i18n("commands.color.color_not_recognised"));
+            return null;
+        }
+
+        String rgbValues = color.getRed() + "," + color.getGreen() + "," + color.getBlue();
+        String hex = getHex(color.getRed(), color.getGreen(), color.getBlue());
+        int decimalColor = Integer.parseUnsignedInt(hex, 16);
+
+        EmbedBuilder builder = MessagingObjects.getMessageTypeEmbedBuilder(MessageType.INFO, context.getUser());
+        builder.setTitle(context.i18n("commands.color.embed_title", hex));
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/utils/ColorUtils.java]: You need to move the lang strings now from commands to utils

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java (hunk 1 above)
Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java]: Don't worry, local/remote mismatches

Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java]: (And you started it)

File: src/main/java/org/cascadebot/cascadebot/utils/ColorUtils.java (hunk 2 above)
Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/utils/ColorUtils.java]: Got it

**Comment by DeadlyFirex** [comment]
@binaryoverload Done <3

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+public class RandomColorSubCommand implements ISubCommand {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        context.reply(ColorUtils.getColor(RandomUtils.randomColor(), context));
+    }
+
+    @Override
+    public String command() { return "color"; }
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java]: For commands we're using the uk spelling for the actual command so this should be `colour`.

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomNumberCommand.java [hunk 4]
```diff
@@ -3,16 +3,20 @@
...
 import org.cascadebot.cascadebot.commandmeta.CommandContext;
 import org.cascadebot.cascadebot.commandmeta.ICommandMain;
+import org.cascadebot.cascadebot.commandmeta.ISubCommand;
 import org.cascadebot.cascadebot.commandmeta.Module;
 import org.cascadebot.cascadebot.permissions.CascadePermission;
 import org.cascadebot.cascadebot.utils.RandomUtils;
 
+import java.util.Set;
+
 public class RandomNumberCommand implements ICommandMain {
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomNumberCommand.java]: This should be renamed to `RandomCommand`

File: src/main/resources/arguments.json [hunk 5]
```diff
@@ -429,5 +429,17 @@
     "amount": {
       "_type": "required"
     }
+  },
+  "random": {
+    "choice": {
```
Comment (weeryan17) [inline src/main/resources/arguments.json]: Both of these should have `*` after them

**Comment by DeadlyFirex** [comment]
Done @weeryan17

**Comment by binaryoverload** [comment]
Sorry, @DeadlyFirex can we revert that change on the UK English? From now on **everything** inside of code files whether that be variables, file names or lang strings should all be in **US** English. The only place for UK English is in the default language file.

Sorry again 😅 This was just me and @weeryan17 being out of sync :)

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import org.cascadebot.cascadebot.utils.RandomUtils;
+
+public class RandomChoiceSubCommand implements ISubCommand {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if (context.getArgs().length > 1) {
+            context.getTypedMessaging().replySuccess(context.i18n("commands.random.random_choice_result", RandomUtils.randomChoice(context.getArgs())));
+        } else {
+            context.getUIMessaging().replyUsage(this);
```
Comment (JamesHawkinss) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java]: I don't like to be picky but having (this) isn't technically necessary, doesn't matter if you choose to leave it in or not.

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Override
+    public String command() { return "choice"; }
+
+    @Override
+    public String parent() { return "random"; }
+
+    @Override
+    public CascadePermission getPermission() { return null; }
```
Comment (JamesHawkinss) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java]: Yas can fill this out `CascadePermission.of(epic stuff);`

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public String command() { return "colour"; }
+
+    @Override
+    public String parent() { return "random"; }
+
+    @Override
+    public CascadePermission getPermission() { return null; }
```
Comment (JamesHawkinss) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomColorSubCommand.java]: Same here :D

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomCommand.java [hunk 9]
```diff
@@ -49,4 +53,9 @@ public CascadePermission getPermission() {
         return CascadePermission.of("random", true);
```
Comment (JamesHawkinss) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/RandomCommand.java]: Why is this in onCommand?

Is it even in onCommand? All the file joins in the GitHub file viewer are throwing me off here :(

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java (hunk 6 above)
Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java]: +1 on this, remove it

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java (hunk 7 above)
Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/RandomChoiceSubCommand.java]: Not needed on this sub command

File: src/main/java/org/cascadebot/cascadebot/commands/fun/RandomCommand.java (hunk 9 above)
Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/RandomCommand.java]: It's not in onCommand ;)

**Comment by DeadlyFirex** [comment]
Reviews done @binaryoverload / @JamesHawkinss , Thanks!

File: src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java [hunk 10]
```diff
@@ -6,99 +6,23 @@
...
-    private static final Pattern BINARY_COLOR = Pattern.compile("([0-1]+)");
-
-    private String getHex(int r, int g, int b) {
-        return String.format("%02x%02x%02x", r, g, b);
-    }
-
     @Override
     public void onCommand(Member sender, CommandContext context) {
         if (context.getArgs().length != 1) {
             context.getUIMessaging().replyUsage(this);
```
Comment (JamesHawkinss) [inline src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: This is that and thus lettuce

Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: ?

Comment (JamesHawkinss) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: Sorry, it says replyUsage(this)

Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: Can you please, for crying out loud, include what is wrong?

Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: > Sorry, it says replyUsage(this)
It should be `replyUsage()`

Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: (Next time)
You left me wondering what the hell you were talking about xD

Comment (DeadlyFirex) [reply src/main/java/org/cascadebot/cascadebot/commands/fun/ColorCommand.java]: By the way, don't blame me for that @JamesHawkinss . It was done by binary

File: src/main/resources/arguments.json [hunk 11]
```diff
@@ -429,5 +429,17 @@
...
     }
+  },
+  "random*": {
+    "choice*": {
+      "words": {
+        "_type": "required"
+      }
+    },
+    "color": {
+    },
```
Comment (weeryan17) [inline src/main/resources/arguments.json]: This needs a type to it. in this case the type would be `command`

**Comment by DeadlyFirex** [comment]
Done @weeryan17

**Comment by JamesHawkinss** [comment]
lovely job, sorry for the confusing review message

**Comment by DeadlyFirex** [comment]
[``](

- 

- [ ] url

)

**Pull request merged**